apps/nextjs-website/src/helpers/user.helper.ts

```diff
@@ -10,18 +11,24 @@ export const useUser = () => {
   const [aligned, setAligned] = useState<boolean>(false);
   const [user, setUser] = useState<DevPortalUser | null>(null);
 
-  const checkUser = useCallback(() => {
-    Auth.currentAuthenticatedUser()
-      .then((user) => {
-        setLoading(false);
-        setAligned(true);
-        setUser(user);
-      })
-      .catch(() => {
-        setLoading(false);
-        setAligned(true);
-        setUser(null);
-      });
+  const checkUser = useCallback(async () => {
```

Comment:
i would call this fetchUserAndSubscriptions
```suggestion
  const fetchUserAndSubscriptions = useCallback(async () => {
```